### DISCUSSION TITLE
Should cue.mod/gen be added to .gitignore?

### DISCUSSION THREAD

**Discussion by MisterMX** [cue-lang/cue#1515] [Q&A] [answered]
Since `cue.mod` is generated via CLI I was wandering if it should be checked in or kept ignored?

Are there any recommendations? I couldn't find anything in the documentation?

**Comment by eonpatapon** [comment, marked as answer]
Well `cue.mod/module.cue` can contain the module name so will need to commit it. Note that `cue` allows the module to be anonymous, so `module.cue` is not strictly required.

However the `cue.mod` directory defines the root of the `cue` module, so I'd say it's better to check it in.

As for `cue.mod/pkg` (the place to put external modules), `cue` doesn't get them automatically yet (see https://github.com/cue-lang/cue/issues/851). So it depends if you need external modules and how you get them.

In theory when package management is implemented it would not be necessary to have `cue.mod/pkg` unless you want to vendor external modules.
